### PR TITLE
updates to overlay web browser

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds a permissions bar and fixes QtWebEngine to work with HTML5 media apis (audio, video, etc) again.  Fixes pop up windows with the main browser  It also adds a version number to our UA for Chrome which makes many more sites functional and bumps the QtQuick version on the main browser to match other controls.

To test:

from "ctrl+b" browser only -- doesnt work with web entities

1. https://dev.windows.com/en-us/microsoft-edge/testdrive/demos/microphone/
2. click start demo
3. you should see the permissions bar.  choose 'allow.' 
4. You should see activity in the signal/frequency graphs
5. if u have a webcam:  https://webcamtoy.com/app/
6. go to www.asana.com.  click login with google.
7. a new window should pop up

### CODE REVIEW

**Comment by huffman** [comment]
I'm uneasy about automatically granting access to audio/webcame/etc. to webpages, this is a security concern.

Thoughts?

**Comment by imgntn** [comment]
last time this came up we decided it was okay.  maybe its time to actually build a proper permissions bar, but that seems separate from fixing this being broken and not working at all.  the default qt web browser implementation seems to have one; maybe we can use it?  http://doc.qt.io/qt-5/qtwebengine-webenginewidgets-demobrowser-featurepermissionbar-h.html

**Comment by imgntn** [comment]
ugh it seems like opening new pop up windows from our browser is broken too; ima try to fix that as well

**Comment by huffman** [comment]
@imgntn If it was a previous decision, then OK. That said, unless we schedule fixing this into our timeline, I have a feeling that will mean we won't address it until something bad happens.

**Comment by imgntn** [comment]
@huffman yeah, i'll make a fogbugz case for adding a permissions bar.  its probably pretty straightforward... i haven't really done a lot of c++/qt UI but i'm down to try

**Comment by hifi-gustavo** [comment]
Build console output for commit: c3c9c0cb10694b10e09ec80c4e0adf627f3dfa79 available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-c3c9c0c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-c3c9c0c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: aacddcf66d360b8a143036e9bb63628d6a566d8c available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-aacddcf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-aacddcf.dmg

**Comment by Nex-Pro** [comment]
### FAIL

Tested on win 10

I only get a pop-up on asana

I get no pop-up for mic or webcam grant permissions.

Also the rendering quality on the web entity  is so bad it hurts my eyes.

There's no address bar so I have to use edit js to enter the address.

The backspace key does nothing in the search box on the Microsoft site.

Forward and back buttons on the mouse don't work

Scroll wheel doesn't work.

Also if my yaw is in the wrong direction all web entities res mirrored left-right

**Comment by imgntn** [comment]
so i think this is a QA pass --you're not supposed to get a pop up for permissions right now.  you should be able to log into asana.  the rest of the stuff below is good feedback, thanks.

**Comment by thoys** [comment]
# QA Testing results

- The audio bar moves, I am also able to get the echo back when I unmute the speaker.
- Managed to get the cam working:
image
- Asana login box pops up correctly.


- Security concerns: YES!
- Noticed a small bug in the browser, the select boxes on the microphone example are offset wrong from the window and appear somewhere outside of the interface window. Not related to this PR.

QA Pass

**Comment by imgntn** [comment]
thanks @thoys 

i'm able to close the browser window every time - with audio/video running- maybe we can get a second opinion?  that's totally weird.

re: security -- agree adding a permissions bar would be a nice feature, maybe you and i can do that.  but since you're the one who opens the browser with Ctrl+B, afaik I can't force one to open on you.

**Comment by SkimiMission** [comment]
## QA Pass

- [SUCCESS] https://dev.windows.com/en-us/microsoft-edge/testdrive/demos/microphone/
- [SUCCESS] 2.You should see activity in the signal/frequency graphs



- [SUCCESS] 3.if u have a webcam: https://webcamtoy.com/app/ 



- [SUCCESS] 4.go to www.asana.com and click on the login
- [SUCCESS] 5.a new window should pop up
new window with google login poped up.



I could close the browser windows without problems. Tested in sandbox.

File: interface/resources/qml/Browser.qml [hunk 1]
```diff
@@ -131,6 +131,9 @@ ScrollingWindow {
             anchors.bottom: parent.bottom
             anchors.left: parent.left
             anchors.right: parent.right
+            onFeaturePermissionRequested: {
+                grantFeaturePermission(securityOrigin, feature, true);
```
Comment (ZappoMan) [inline interface/resources/qml/Browser.qml]: I think you should use the exist OK/Cancel or Yes/No dialog box we have and simply pop up a request to grant the permission and let the user decide. That's the better approach to this.

**Comment by imgntn** [comment]
working on a permissions bar

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8acb30da8e1f19d8c906539cd0e16b1280e1fca2 available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-8acb30d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-8acb30d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3b25b36db32ee5586b8fb98e1e2ec0195eb91470 available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-3b25b36.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-3b25b36.dmg

**Comment by imgntn** [comment]
added a permissions bar and updated the test plan above

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea1c21e685cd37fc716e1b101997fd11b2088dee available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-ea1c21e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-ea1c21e.dmg

**Comment by SkimiMission** [comment]
## QA Pass

- [SUCCESS] 1.https://dev.windows.com/en-us/microsoft-edge/testdrive/demos/microphone/
- [SUCCESS] 2.click start demo
- [SUCCESS] 3.you should see the permissions bar. choose 'allow.' 
- [SUCCESS] 4.You should see activity in the signal/frequency graphs
- [SUCCESS] 5.if u have a webcam: https://webcamtoy.com/app/ 
- [SUCCESS] 6.go to www.asana.com. click login with google.
- [SUCCESS] 7.a new window should pop up

also closing of the web browser works fine for me.

File: interface/resources/qml/Browser.qml [hunk 2]
```diff
@@ -117,21 +132,84 @@ ScrollingWindow {
...
+                            root.hidePermissionsBar();
                             webview.url = text
                             break;
                     }
                 }
             }
         }
 
+        Rectangle {
+        id:permissionsContainer
```
Comment (ZappoMan) [inline interface/resources/qml/Browser.qml]: lines 144-154 should be indented one level

File: interface/resources/qml/Browser.qml [hunk 3]
```diff
@@ -117,21 +132,84 @@ ScrollingWindow {
...
             id: webview
-            url: "http://highfidelity.com"
+            url: "https://highfidelity.com"
             anchors.top: buttons.bottom
             anchors.topMargin: 8
             anchors.bottom: parent.bottom
             anchors.left: parent.left
             anchors.right: parent.right
+            onFeaturePermissionRequested: {
+                permissionsBar.securityOrigin =securityOrigin;
```
Comment (ZappoMan) [inline interface/resources/qml/Browser.qml]: weird spacing on this and next line... let's follow same coding standard as C++/JS for this... `foo = bar;`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 38765e6fa4b59e1f14ac79f800d09dd0916b5242 available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-38765e6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-38765e6.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84e4de4bad8043136a560d6ad855155a9e12dde8 available at: https://gist.github.com/ac368ab3afb72efbdeed0d59a0d6d7ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-84e4de4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8237-84e4de4.dmg

**Pull request merged**